storm-core/src/jvm/org/apache/storm/utils/Utils.java

```diff
@@ -1382,5 +1422,438 @@ public static TopologyInfo getTopologyInfo(String name, String asUser, Map storm
+             files.
+     */
+    public static Map mapVal (IFn aFn, Map amap) {
+        Map newMap = new HashMap();
+        if (amap == null) {
+            return newMap;
+        }
+        if (amap.keySet()==null) {
+            return newMap;
+        }
+        for (Object key: amap.keySet()) {
+            Object value = amap.get(key);
+            if (value == null) {
+                newMap.put(key, null);
+            } else {
+                Object newValue = aFn.eval(value);
+                newMap.put(key, newValue);
+            }
+        }
+        return newMap;
+    }
+
+    /* TODO: This function was originally written to replace map-val in util.clj. But we decided to change the coding
+         style in the caller functions to a more Java style for loop. This is mentioned in TODOs across the clojure
+         files.
+    */
+//    public static Map filterVal(IPredicate aFn, Map amap) {
+//        Map newMap = new HashMap();
+//        if (amap == null) {
+//            return newMap;
+//        }
+//        for (Object key: amap.keySet()) {
+//            Object value = amap.get(key);
+//            if(aFn.test(value)) {
+//                newMap.put(key, value);
+//            }
+//        }
+//        return newMap;
+//    }
+
+    /* TODO: This function was originally written to replace filter-key in util.clj. But we decided to change the coding
+         style in the caller functions to a more Java style for loop + if conditionals. This is mentioned in TODOs
+         across the clojure files.
+    */
+//    public static Map filterKey(IPredicate aFn, Map amap) {
+//        Map newMap = new HashMap();
+//        if (amap == null) {
+//            return newMap;
+//        }
+//        for (Object key: amap.keySet()) {
+//            Object value = amap.get(key);
+//            if(aFn.test(key)) {
+//                newMap.put(key, value);
+//            }
+//        }
+//        return newMap;
+//    }
+
+    /* TODO: This function was originally written to replace map-key in util.clj. But we decided to change the coding
+         style in the caller functions to a more Java style for loop. This is mentioned in TODOs across the clojure
+         files.
+    */
+//    public static Map mapKey (IFn aFn, Map amap) {
+//        Map newMap = new HashMap();
+//        if (amap == null) {
+//            return newMap;
+//        }
+//        for (Object key: amap.keySet()) {
+//            Object value = amap.get(key);
+//            Object newKey = aFn.eval(key);
+//            newMap.put(newKey, value);
+//        }
+//        return newMap;
+//    }
+
+    /* TODO: This function was originally written to replace separate in util.clj. But since it was only used in
+    transactional_test.clj, the separate function was moved there for now.
+    */
+    public static Vector<Collection> separate (IPredicate pred, Collection aseq) {
```

Comment:
I don't really want commented out code, or dead code.  Can we please delete it unless we are actually using it.